cmake/rust_bridge.cmake

```diff
@@ -128,7 +161,18 @@ if("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
     set(Rust_CARGO_TARGET "x86_64-unknown-linux-gnu")
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
-    set(Rust_CARGO_TARGET "x86_64-apple-darwin")
+    # on macOS "uname -m" returns the architecture (x86_64 or arm64)
+    execute_process(
+        COMMAND uname -m
+        RESULT_VARIABLE exit_code_or_error
+        OUTPUT_VARIABLE OSX_NATIVE_ARCHITECTURE
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    if(OSX_NATIVE_ARCHITECTURE STREQUAL "arm64")
+        set(Rust_CARGO_TARGET "aarch64-apple-darwin")
+    else()
+        set(Rust_CARGO_TARGET "x86_64-apple-darwin")
+    endif()
```

Comment:
Use `uname -m` to detect architecture and set the target correctly (so that apple silicon can work)